server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java

```diff
@@ -113,15 +123,18 @@ public String toString() {
 
         final Releasable releasable = coordinator.withDiscoveryListener(respondIfRequestSatisfied);
         listenableFuture.addListener(ActionListener.wrap(releasable::close), directExecutor, threadPool.getThreadContext());
-        respondIfRequestSatisfied.accept(coordinator.getFoundPeers());
+
+        if (coordinator.isInitialConfigurationSet()) {
+            respondIfRequestSatisfied.onFailure(new ClusterAlreadyBootstrappedException());
+        } else {
+            respondIfRequestSatisfied.onResponse(coordinator.getFoundPeers());
+        }
 
         if (request.getTimeout() != null) {
             threadPool.schedule(request.getTimeout(), Names.SAME, new Runnable() {
                 @Override
                 public void run() {
-                    if (listenerNotified.compareAndSet(false, true)) {
```

Comment:
why is this not needed anymore?

Reply:
`respondIfRequestSatisfied.onFailure()` checks this.

Reply:
ok thanks